packages/table/components/CheckboxTable.tsx

```diff
@@ -52,7 +58,7 @@ class CheckboxTable extends React.PureComponent<CheckboxTableProps> {
           id={rowIndex.toString()}
           inputLabel={`Toggle row ${rowIndex}`}
           showInputLabel={false}
-          checked={selectedRows.includes(rowData)}
+          checked={selectedRows.includes(rowId)}
```

Comment:
Would an object also work so we could replace `includes` with just `selectedRows[rowId]` and `filter` with `delete selectedRows[rowId]`. Is there any specific reason we would like having them as array?

Reply:
I don't see a reason why not. An array just felt more appropriate for a list of things.

So you're imagining an object that would look something like:
```
{
    row1: true,
    row2: true,
    ...etc
}
```

Reply:
exactly, if it's just a helper data structure anyway

Reply:
Though it won't be possible for _all_ types that `const rowId = rowData[uniqueKey];` can return. 
`rowData[uniqueKey]` could be an `object` right? But then again we can't compare objects anyway.

Reply:
Yes, it _could_ return an object, but I'm trusting consumers would only use a piece of row data that's a `number` or `string`. I should add that detail in the prop documentation...